### ISSUE TITLE
All machine readable output "Lists" should use Kind: List

### ISSUE THREAD

**Issue by cdrage** [openshift/odo#2220]
[kind/Enhancement]

## Which functionality do you think we should update/improve?

There are some json machine output commands that are not using the correct List for example:

- `odo catalog list services` uses ServiceTypeList
- `odo catalog list components` uses ComponentTypeList
- `odo application list` uses ApplicationList

These should be `List`

## Why is this needed? 

To follow Kubernetes standards.

**Comment by mohammedzee1000** [comment]
This issue is marked for clean up, so lets get that removed or it will close

**Comment by kadel** [comment]
> These should be `List`

What `List`?
@cdrage point to the place where is being done correctly?

**Comment by dev-gaur** [comment]
I think maybe @cdrage meant something on the lines of https://github.com/kubernetes/kubectl/blob/9fae31af80294050f1480951b7fae1ababad9c1a/pkg/cmd/get/get.go#L764 .

I had been thinking that maybe this issue asks me to remove the types like project/ProjectLIst , component/ComponentList, catalog/ServiceTypeList etc... ( I think I got confused by the `kind/cleanup` label on this issue. )  and then use the k8s.io/apimachinery/corev1.List or metav1.List  type wherever those List types are being computed.

Now, after reading "readable output" in the Issue's title, maybe the intention is to just the print resources Lists ... as  k8s.io/apimachinery/corev1.List or metav1.List instead of our self defined types like ProjectList , AppList, etc...

I was confused why should these List types be cleaned up since these are being used in other standard projects like k8s/client-go, openshift/client-go and kubectl as well..but then I realised maybe this is only about the aspect of machine readable output and not all other processing logic.

I might be wrong even now but this is my fresh look on this issue.
